src/app/pages/checkout-payment/payment-payone-creditcard/payment-payone-creditcard.component.ts

```diff
@@ -0,0 +1,165 @@
+
+  private destroy$ = new Subject();
+
+  /**
+   * flag to make sure that the init script is executed only once
+   */
+  scriptLoaded = false;
+
+  // tslint:disable-next-line: no-any
+  iframes: any;
+
+  ngOnInit() {
+    this.payoneCreditCardForm = new FormGroup({
+      saveForLater: new FormControl(true),
+    });
+    // keep reference for payone cc component
+    const thisComp = this;
+
+    // register helper function to call the callback function of this component
+    // tslint:disable-next-line:no-string-literal only-arrow-functions no-any
+    (window as any)['payoneCreditCardCallback'] = function (response: {
+      status: string;
+      pseudocardpan: string;
+      truncatedcardpan: string;
+    }) {
+      thisComp.submitCallback(response);
+    };
+  }
+
+  /**
+   * load payone script if component is shown
+   */
+  ngOnChanges() {
+    if (this.paymentMethod) {
+      this.loadScript();
+    }
+  }
+
+  ngOnDestroy() {
+    // unregister helper function again
+    // tslint:disable-next-line:no-string-literal no-any
+    (window as any)['payoneCreditCardCallback'] = undefined;
+
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  protected getParamValue(name: string, errorMessage: string): string {
+    const parameter = this.paymentMethod.hostedPaymentPageParameters.find(param => param.name === name);
+    if (!parameter || !parameter.value) {
+      this.payoneCreditCardForm.controls.error.setValue(errorMessage);
+      return;
+    }
+    return parameter.value;
+  }
+
+  loadScript() {
+    // load script only once if component becomes visible
+    if (this.activated && !this.scriptLoaded) {
+      const request = JSON.parse(this.getParamValue('request', 'checkout.credit_card.param.error.notFound'));
+      const config = JSON.parse(this.getParamValue('config', 'checkout.credit_card.param.error.notFound'));
+
+      this.scriptLoaded = true;
+      this.scriptLoader
+        .load('https://secure.pay1.de/client-api/js/v1/payone_hosted_min.js')
+        .pipe(takeUntil(this.destroy$))
+        .subscribe(() => {
+          // append localization language: Payone.ClientApi.Language.en, Language to display error-messages (default:Payone.ClientApi.Language.en)
+          const avoidEvalCheck = eval;
+          config.language = avoidEvalCheck(this.getParamValue('language', ''));
+
+          // setup
+          this.iframes = new Payone.ClientApi.HostedIFrames(config, request);
+        });
+    }
+  }
+
+  /**
+   * cancel new payment instrument, hides and resets the parameter form
+   */
+  cancelNewPaymentInstrument() {
+    this.payoneCreditCardForm.reset();
+    this.cancel.emit();
+  }
+
+  /**
+   * submit payone payment form
+   */
+  submitNewPaymentInstrument() {
+    if (this.iframes.isComplete()) {
+      // Perform "CreditCardCheck" to create and get a PseudoCardPan; then call your function "payCallback"
+      this.iframes.creditCardCheck('payoneCreditCardCallback');
+    }
```

Comment:
If the user clicks submit and the iframe is incomplete nothing happens. In this case error messages should be shown and the empty input fields should be marked red. - Furthermore I cannot see any error handling by now.

Reply:
All fields are IFRAMES and we can not access the data for validation. Payone automatically does error handling (Please refer to screenshot for example). In case of incomplete data, the Payone scripts do not trigger any call.
![PayoneErrorHandling](https://user-images.githubusercontent.com/67738151/125910748-23171b88-911b-4982-90b7-84611f8470e7.JPG)

Reply:
Pls, add the following error message in case the form is incomplete and the user clicked submit: 
"The form is incomplete. Please fill in all fields marked as required and submit the form to continue."
If the form is complete and the user clicks submit this message should be removed.
The error message should have the same appearance as the error message above.